src/PKSim.Core/Services/HIDiseaseStateImplementation.cs

```diff
@@ -0,0 +1,232 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Core.Domain.UnitSystem;
+using OSPSuite.Utility.Collections;
+using OSPSuite.Utility.Extensions;
+using PKSim.Assets;
+using PKSim.Core.Model;
+using PKSim.Core.Repositories;
+using static OSPSuite.Core.Domain.Constants.Parameters;
+using static PKSim.Core.CoreConstants.Units;
+using static PKSim.Core.CoreConstants.Organ;
+using static PKSim.Core.CoreConstants.Parameters;
+using HIFactors = OSPSuite.Utility.Collections.Cache<double, double>;
+using IFormulaFactory = OSPSuite.Core.Domain.Formulas.IFormulaFactory;
+using IParameterFactory = PKSim.Core.Model.IParameterFactory;
+
+namespace PKSim.Core.Services
+{
+   public class HIDiseaseStateImplementation : AbstractDiseaseStateImplementation
+   {
+      private readonly IParameterFactory _parameterFactory;
+
+      public static class ChildPughScore
+      {
+         public static double A = 0;
+         public static double B = 1;
+         public static double C = 2;
+      }
+
+      private static HIFactors createFactors(double forA, double forB, double forC) =>
+         new()
+         {
+            {ChildPughScore.A, forA},
+            {ChildPughScore.B, forB},
+            {ChildPughScore.C, forC},
+         };
+
+      private static readonly HIFactors _portalFlowScalingFactor = createFactors(0.4, 0.36, 0.04);
+      private static readonly HIFactors _hepaticFlowScalingFactor = createFactors(1.3, 2.3, 3.4);
+      private static readonly HIFactors _hepaticVolumeScalingFactor = createFactors(0.69, 0.55, 0.28);
+      private static readonly HIFactors _renalFlowScalingFactor = createFactors(0.88, 0.65, 0.65);
+      private static readonly HIFactors _otherOrgansFlowScalingFactor = createFactors(1.31, 1.84, 2.27);
+      private static readonly HIFactors _gfrScalingFactor = createFactors(1, 0.7, 0.36);
+      private static readonly HIFactors _albuminFactor = createFactors(0.81, 0.68, 0.5);
+      private static readonly HIFactors _agpFactor = createFactors(0.6, 0.56, 0.3);
+      private static readonly HIFactors _hematocritScalingFactor = createFactors(0.92, 0.88, 0.83);
+
+      private static readonly Cache<string, HIFactors> _moleculeScalingFactorEdginton = new()
+      {
+         {"CYP3A4", createFactors(1, 0.4, 0.4)},
+         {"CYP1A2", createFactors(1, 0.1, 0.1)},
+         {"CYP2E1", createFactors(1, 0.83, 0.83)},
+      };
+
+      private static readonly Cache<string, HIFactors> _moleculeScalingFactorJohnson = new()
```

Comment:
why do we need 2 different caches for the molecule scaling factors?

Reply:
good points. Two different sources, but unique key

Reply:
we need two otherwise I do not know what origin data to use